framework/wazuh/decoder.py

```diff
@@ -290,45 +332,53 @@ def upload_decoder_file(filename: str, content: str, overwrite: bool = False) ->
         result.affected_items.append(to_relative_path(full_path))
         result.total_affected_items = len(result.affected_items)
         backup_file and exists(backup_file) and remove(backup_file)
-    except WazuhError as e:
-        result.add_failed_item(id_=to_relative_path(full_path), error=e)
+    except WazuhError as exc:
+        result.add_failed_item(id_=to_relative_path(full_path), error=exc)
     finally:
         exists(backup_file) and safe_move(backup_file, full_path)
 
     return result
 
 
 @expose_resources(actions=['decoders:delete'], resources=['decoder:file:{filename}'])
-def delete_decoder_file(filename: str) -> AffectedItemsWazuhResult:
+def delete_decoder_file(filename: str, relative_dirname: str = None) -> AffectedItemsWazuhResult:
     """Delete a decoder file.
 
     Parameters
     ----------
     filename : str
         Name of the decoder file.
-
+    relative_direname : str
+        Relative directory where de decoder is found.
+        
     Returns
     -------
     AffectedItemsWazuhResult
         Affected items.
     """
-    result = AffectedItemsWazuhResult(all_msg='Decoder file was successfully deleted',
-                                      none_msg='Could not delete decoder file'
-                                      )
+    file = filename[0] if isinstance(filename, list) else filename
 
-    full_path = join(common.USER_DECODERS_PATH, filename[0])
+    result = AffectedItemsWazuhResult(all_msg='Decoder file was successfully deleted',
+                                      none_msg='Could not delete decoder file')
+    # Validate if the relative_dir is a decoder_dir
+    ruleset_conf = configuration.get_ossec_conf(section='ruleset')['ruleset']
+    relative_dirname = relative_dirname.rstrip('/') if relative_dirname \
+        else to_relative_path(common.USER_DECODERS_PATH)
+    if not relative_dirname in ruleset_conf['decoder_dir']:
+        raise WazuhError(1505)
 
+    full_path = join(common.WAZUH_PATH, relative_dirname, file)  
     try:
         if exists(full_path):
             try:
                 remove(full_path)
                 result.affected_items.append(to_relative_path(full_path))
-            except IOError:
-                raise WazuhError(1907)
+            except IOError as exc:
+                raise WazuhError(1907) from exc
```

Comment:
Here we should use the `extra_message` parameter from `WazuhError` if we want to include more information about the error. I'm not sure if `from` has any impact if used like this

Reply:
The "from exc" establishes the cause relationship between exceptions is established. By using "from exc", it indicates that the current exception is a direct consequence of the exception represented by exc. The "from exc" construction is also recommended by Pylint.

When this exception is raised, Python will display the traceback, showing the sequence of function calls that led to the exception. In the traceback, the previous exception (exc) will be shown, followed by the current exception (WazuhError). This can be useful for debugging and understanding the sequence of events that led to the error.

Reply:
I got you, I was referring to the information that is displayed to the user via the API but I omitted the fact that it is usefult for debugging purposes. I think it's a good idea to start nesting errors to get a better understanding of the errors we get during development.